plugin/plugin.go

```diff
@@ -0,0 +1,396 @@
+	logrus.Infof("Number of readable files: %d", len(validFiles))
+
+	if len(validFiles) == 0 {
+		return nil, errors.New("no readable files found matching the report filename pattern")
+	}
+
+	return validFiles, nil
+}
+
+// processFile reads a TestNG XML report using xml.Decoder for streaming, validates its structure, and logs details.
+func processFile(filename string) (Results, error) {
+	logrus.Infof("Processing file: %s", filename)
+
+	// Open the file for streaming
+	file, err := os.Open(filename)
+	if err != nil {
+		if os.IsNotExist(err) {
+			logrus.Errorf("File not found: %s", filename)
+			return Results{}, fmt.Errorf("file not found: %s", filename)
+		}
+		if os.IsPermission(err) {
+			logrus.Errorf("Permission denied for file: %s", filename)
+			return Results{}, fmt.Errorf("permission denied for file: %s", filename)
+		}
+		logrus.Errorf("Error opening file: %s. Error: %v", filename, err)
+		return Results{}, fmt.Errorf("error opening file: %s. Error: %v", filename, err)
+	}
+	defer file.Close()
+
+	// Use xml.Decoder for streaming
+	decoder := xml.NewDecoder(file)
+	var report TestNGReport
+
+	if err := decoder.Decode(&report); err != nil {
+		logrus.WithError(err).WithField("File", filename).Error("Failed to parse TestNG XML")
+		return Results{}, fmt.Errorf("failed to parse TestNG XML for file: %s. Error: %v", filename, err)
+	}
+
+	// Validate structure
+	if len(report.Suites) == 0 {
+		logrus.Infof("File %s contains no test suites in the XML structure", filename)
+		return Results{}, fmt.Errorf("no test suites found in the XML structure of file: %s", filename)
+	}
+
+	for _, suite := range report.Suites {
+		if len(suite.Classes) == 0 {
+			logrus.Infof("Suite '%s' in file %s contains no test classes", suite.Name, filename)
+		}
+	}
+
+	// Log details and return results
+	return logTestNGReportDetails(report), nil
+}
+
+// logTestNGReportDetails logs the details of a TestNG report and returns the aggregated results.
+func logTestNGReportDetails(report TestNGReport) Results {
+	results := Results{}
+	var failedTests []string
+	var skippedTests []string
+
+	// Aggregate data across all suites
+	for _, suite := range report.Suites {
+		suiteResults, failed, skipped := aggregateSuiteResults(suite)
+		results.Total += suiteResults.Total
+		results.Failures += suiteResults.Failures
+		results.Skipped += suiteResults.Skipped
+		results.DurationMS += suiteResults.DurationMS
+
+		failedTests = append(failedTests, failed...)
+		skippedTests = append(skippedTests, skipped...)
+
+		// Log suite summary
+		logSuiteSummary(suite.Name, suiteResults)
+		// Log groups and test details
+		logSuiteGroups(suite)
+		logSuiteTestDetails(suite)
+	}
+
+	// Log aggregated results with failed and skipped test names
+	if len(failedTests) > 0 {
+		logrus.Infof("\nTest case Failures: %s", formatTestNames(failedTests))
+	}
+	if len(skippedTests) > 0 {
+		logrus.Infof("\nTest case Skips: %s", formatTestNames(skippedTests))
```

Comment:
```suggestion
		logrus.Infof("\nFailed Test cases: %s", formatTestNames(failedTests))
	}
	if len(skippedTests) > 0 {
		logrus.Infof("\nSkipped Test cases: %s", formatTestNames(skippedTests))
```
<img width="818" alt="Screenshot 2024-12-17 at 8 56 54 PM" src="https://github.com/user-attachments/assets/1df8ee4c-b712-4e67-9c72-176929308e27" />
add `======` before `Failed Test cases` for clarity